ecosystem/node-checker/src/evaluators/direct/api/transaction_presence.rs

```diff
@@ -0,0 +1,195 @@
+    /// baseline produced after a delay. We confirm that the transactions are
+    /// same by looking at the version.
+    async fn evaluate(&self, input: &Self::Input) -> Result<Vec<EvaluationResult>, Self::Error> {
+        let baseline_client =
+            AptosRestClient::new(input.baseline_node_information.node_address.get_api_url());
+
+        let latest_baseline_transaction_info = Self::get_transaction_info(&baseline_client).await?;
+
+        tokio::time::sleep(Duration::from_secs(self.args.transaction_fetch_delay_secs)).await;
+
+        let target_client = AptosRestClient::new(input.target_node_address.get_api_url());
+        let evaluation = match Self::get_transaction_info_from_hash(
+            &target_client,
+            HashValue::from(latest_baseline_transaction_info.hash),
+        )
+        .await
+        {
+            Ok(latest_target_transaction_info) => {
+                if latest_baseline_transaction_info.accumulator_root_hash
+                    == latest_target_transaction_info.accumulator_root_hash
+                {
+                    self.build_evaluation_result(
+                        "Target node produced valid recent transaction".to_string(),
+                        100,
+                        format!(
+                            "We got the latest transaction from the baseline node ({}), waited {} \
+                                seconds, and then asked your node to give us that transaction, and \
+                                it did. Great! This implies that your node is keeping up with other \
+                                nodes in the network.",
+                            latest_baseline_transaction_info.hash, self.args.transaction_fetch_delay_secs,
+                        ),
+                    )
+                } else {
+                    self.build_evaluation_result(
+                        "Target node produced recent transaction, but it was invalid".to_string(),
+                        75,
+                        format!(
+                            "We got the latest transaction from the baseline node ({}), waited {} \
+                                seconds, and then asked your node to give us that transaction, and \
+                                it did. However, the transaction was invalid compared to the baseline \
+                                as the version of the transaction ({}) was different compared to \
+                                the baseline ({}).",
+                            latest_baseline_transaction_info.hash,
+                            self.args.transaction_fetch_delay_secs,
+                            latest_target_transaction_info.accumulator_root_hash,
+                            latest_baseline_transaction_info.accumulator_root_hash,
+                        ),
+                    )
+                }
+            }
+            Err(e) => self.build_evaluation_result(
```

Comment:
It's not great that we're grouping two cases:
1. Invalid data
2. Missing data

Invalid data seems much more dangerous to me and we should be scoring 0, while missing data may be a result of a super aggressive pruner, or the node not having synced to the version being requested. Is it possible to split these two cases up here?